### ISSUE TITLE
local_auth plugin wrongfully reports available methods when they're not enabled on the device.

### ISSUE THREAD

**Issue by ptheofan** [flutter/flutter#77799]
## Steps to Reproduce
1. Create an Android Pixel device and **DO NOT** initialize the fingerprint sensor.
2. Download the local_auth example code and run it
3. Click the get methods listing, you will get fingerprint method available (which is not true since is not initialized by the user on the device yet)
4. Click the button to Authenticate - exception.

**Expected results:**
Getting available methods listing should be empty since as the name implies it's available methods and the device currently has NO methods readily available. Currently, it basically lists hardware device capabilities as methods.

**Actual results:**
I get the fingerprint method as available.

<details>
  <summary>Logs</summary>
[        ] 💪 Running with sound null safety 💪
[+16590 ms] I/flutter (10813): PlatformException(NotAvailable, Required security features not enabled, null, null)

</details>

**Comment by bharat-biradar** [comment]
Hi @TahaTesser I would like to work on this issue.

I was thinking of this approach to solve this issue.
Keeping the `getAvailableBiometrics()` the same which returns the hardware supported features, 
there could be another function `getEnrolledBiometrics()`  which can return the list of enrolled biometrics.

I wanted to use [Androidx Biometric](https://developer.android.com/reference/androidx/biometric/package-summary) library to get whether the user has enrolled or not. But the problem with using this library is we can only determine that user has enrolled and also whether device supports to either [BIOMETRIC_STRONG](https://developer.android.com/reference/androidx/biometric/BiometricManager.Authenticators#BIOMETRIC_STRONG) etc.. So should the function return the type of biometrics enrolled or the different levels of security that device has been enrolled in.

For api<27 we can use `FingerprintManager` to to check whether there is a fingerprint enrolled but the api is deprecated.

Please let me what should be done. Thanks.

**Comment by bharat-biradar** [comment]
Can I open PR for the changes I have made?

**Comment by ptheofan** [comment]
> Hi @TahaTesser I would like to work on this issue.
> 
> I was thinking of this approach to solve this issue.
> Keeping the `getAvailableBiometrics()` the same which returns the hardware supported features,
> there could be another function `getEnrolledBiometrics()` which can return the list of enrolled biometrics.
> 
> I wanted to use [Androidx Biometric](https://developer.android.com/reference/androidx/biometric/package-summary) library to get whether the user has enrolled or not. But the problem with using this library is we can only determine that user has enrolled and also whether device supports to either [BIOMETRIC_STRONG](https://developer.android.com/reference/androidx/biometric/BiometricManager.Authenticators#BIOMETRIC_STRONG) etc.. So should the function return the type of biometrics enrolled or the different levels of security that device has been enrolled in.
> 
> For api<27 we can use `FingerprintManager` to to check whether there is a fingerprint enrolled but the api is deprecated.
> 
> Please let me what should be done. Thanks.

Actually, the `getAvailableBiometrics` is intended to return the valid (enabled) methods. This is reflected also in the iOS implementation. It will return the method such as `faceid` only after the user has enabled it.

Would be very nice to have a different function to retrieve a list of the platform supported methods although this is outside the scope of this ticket.

**Comment by bharat-biradar** [comment]
Actually I scraped the idea of creating another function and modified `getAvailableBiometrics` to retrieve only those which are enabled.

**Comment by alienspaces** [comment]
Hi there, is there a timeline for when this might be made available in a release?

**Comment by stuartmorgan** [comment]
Duplicate of https://github.com/flutter/flutter/issues/45497
